conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -339,6 +342,28 @@ private TypeSpec generateEndpointHandler(
         return endpointBuilder.build();
     }
 
+    private static final ClassName LIST_NAME = ClassName.get(List.class);
+    private static final ClassName IMMUTABLE_LIST_NAME = ClassName.get(ImmutableList.class);
+    private static final ClassName SET_NAME = ClassName.get(Set.class);
+    private static final ClassName IMMUTABLE_SET_NAME = ClassName.get(ImmutableSet.class);
+
+    private TypeName immutableCollection(TypeName input) {
+        // Note that only the outermost collection is considered for replacement to avoid
+        // generic type incompatibilities.
+        if (options.nonNullCollections() && input instanceof ParameterizedTypeName) {
```

Comment:
@tusharnarayan can you remind me why your service (f/c ?) doesn't currently use nonNullCollections?

Reply:
We persist Conjure objects with collections in a number of places. It's difficult for us to be confident that nulls have not been persisted.

Reply:
The path forward is for us to replace Conjure object in our storage layer with out own objects that allow nulls, but this is a non-trivial undertaking.

Reply:
For context, see f/c/pull/9962.

Reply:
We'll just pick up the work to audit and deal with any stored Conjure objects, and then will turn on nonNullCollections - once this PR is merged, should solve the case I outlined in https://github.com/palantir/conjure-java/issues/1050